cfme/tests/infrastructure/test_vm_retirement_rest.py

```diff
@@ -1,43 +1,31 @@
 # -*- coding: utf-8 -*-
 import datetime
 import pytest
-import fauxfactory
 
 from cfme import test_requirements
 from cfme.rest.gen_data import a_provider as _a_provider
-from utils.version import current_version
-from utils.virtual_machines import deploy_template
+from cfme.rest.gen_data import vm as _vm
 from utils.wait import wait_for
 
 
 pytestmark = [test_requirements.retirement]
 
 
-@pytest.fixture(scope='function')
+@pytest.fixture(scope="function")
 def a_provider():
     return _a_provider()
 
 
-@pytest.fixture(scope='function')
+@pytest.fixture(scope="function")
 def vm(request, a_provider, rest_api):
-    # Don't use cfme.rest.gen_data.vm because we don't need finalizer and delete vm after test
-    provider_rest = rest_api.collections.providers.get(name=a_provider.name)
-    vm_name = deploy_template(
-        a_provider.key,
-        "test_rest_vm_{}".format(fauxfactory.gen_alphanumeric(length=6)))
-    provider_rest.action.refresh()
-    wait_for(
-        lambda: len(rest_api.collections.vms.find_by(name=vm_name)) > 0,
-        num_sec=600, delay=5)
-    return vm_name
+    return _vm(request, a_provider, rest_api)
 
 
 @pytest.mark.tier(3)
-@pytest.mark.uncollectif(lambda: current_version() <= "5.5.2.4")
 @pytest.mark.parametrize(
-    "multiple", [True, False],
+    "from_collection", [True, False],
```

Comment:
**FUTURE:** I am not 100% loving the fixture name here. We should call it something like ```source```, the only reason I say this is that ```from_collection``` conflicts with the actual ID ```from_collection``` yielding confusion around having two different IDs that always provide the same fixture, and one of them matches the name.......gah that sounded terrible....hence my confusion initially. I'd considering changing it to ```method``` or ```source``` but not in this PR.